server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java

```diff
@@ -801,15 +801,21 @@ private void startCleaner() {
                             if (result.bytes() > 0L || result.blobs() > 0L) {
                                 // we could legitimately run into dangling blobs as the result of a shard snapshot failing half-way
                                 // through the snapshot because of a concurrent index-close or -delete. The second round of cleanup on
-                                // the same repository however must always fully remove any dangling blobs since we block all concurrent
-                                // operations on the repository here
+                                // the same repository however should always find no more dangling blobs and be a no-op since we block all
+                                // concurrent operations on the repository.
                                 client.admin()
                                     .cluster()
                                     .prepareCleanupRepository(trackedRepository.repositoryName)
                                     .execute(mustSucceed(secondCleanupRepositoryResponse -> {
                                         final RepositoryCleanupResult secondCleanupResult = secondCleanupRepositoryResponse.result();
-                                        assertThat(Strings.toString(secondCleanupResult), secondCleanupResult.blobs(), equalTo(0L));
-                                        assertThat(Strings.toString(secondCleanupResult), secondCleanupResult.bytes(), equalTo(0L));
+                                        if (secondCleanupResult.blobs() == 1) {
```

Comment:
Should we assert that the blob name starts with `index-`?

Reply:
There isn't really an easy way to do that AFAICT, at least not without adding a bunch more infrastructure. And this isn't a very important assertion in the context of these tests.